packages/slate/src/changes/at-range.js

```diff
@@ -990,7 +990,7 @@ Changes.splitBlockAtRange = (change, range, height = 1, options = {}) => {
 
   if (range.isExpanded) {
     change.deleteAtRange(range, { normalize })
-    range = range.collapseToStart()
+    range = change.value.selection
```

Comment:
Sometimes, start block may be deleted. So that we cannot just collapse selection to start, but get the new selection from changed value.

Reply:
I think as written here this will only work when the `range` to be deleted is also the currently selected range (ie. `range === selection`).

Instead, it seems like we'd need to check if the start key still exists in the document after the delete, and if it doesn't then select the key of the text node before it?

Reply:
Actually, this work even though the start block is not been deleted.

> Instead, it seems like we'd need to check if the start key still exists in the document after the delete, and if it doesn't then select the key of the text node before it?

`change.deleteAtRange` had did this for us, so what we need to do is just read the latest selection.

Reply:
The problem here is that `*AtRange` methods doesn't act on the current user's selection, they act at an arbitrary range in the document. So they can't use `change.value.selection`, because someone may be splitting a block at a range programmatically that isn't the current selection.

Reply:
OK, I understand.